### PR TITLE
docs(multiple): revise and expand a11y docs

### PR DESCRIPTION
**Pull request by jelbourn**
This change updates the accessibility docs for 21 components for
correctness, comprehensiveness, and clarity.

I have intentionally omitted several other components either because
they're being worked on by someone else or have significant changes
pending.

cc @zarend @amysorto

### CODE REVIEW

File: src/material/autocomplete/autocomplete.md [hunk 1]
```diff
@@ -112,8 +114,16 @@ autocomplete is attached to using the `matAutocompleteOrigin` directive together
               "region":"mat-autocomplete"}) -->
 
 ### Accessibility
-The input for an autocomplete without text or labels should be given a meaningful label via
-`aria-label` or `aria-labelledby`.
 
-The autocomplete trigger is given `role="combobox"`. The trigger sets `aria-owns` to the
-autocomplete's id, and sets `aria-activedescendant` to the active option's id.
+`MatAutocomplete` implements the ARIA combobox interaction pattern. The text input trigger specifies
+`role="comboxbox"` while the content of the pop-up applies `role="listbox"`. Because of this listbox
```
Comment (crisbeto) [inline src/material/autocomplete/autocomplete.md:119]: comboxbox -> combobox

File: src/material/datepicker/datepicker.md [hunk 2]
```diff
@@ -562,14 +562,22 @@ value can be anything that is accepted by `ngClass`.
...
-`MatDatepickerIntl` includes strings that are used for `aria-label`s. The datepicker input
-should have a placeholder or be given a meaningful label via `aria-label`, `aria-labelledby` or
+The `MatDatepickerInput` and `MatDatepickerToggle` directives both apply the `aria-haspopup`
+attribute to the native input and button elements, respectively.
+
+`MatDatepickerIntl` includes strings that are used for `aria-label` attributes. Always provide
+the datepicker text input a meaningful label via `<mat-label>`, `aria-label`, `aria-labelledby` or
 `MatDatepickerIntl`.
 
+`MatDatepickerInput` adds `Alt + Down Arrow` as a keyboard short to open the datepicker pop-up.
```
Comment (crisbeto) [inline src/material/datepicker/datepicker.md:576]: Should the keyboard shortcut use `<kbd>`?

File: src/material/tabs/tabs.md [hunk 3]
```diff
@@ -85,22 +85,36 @@ duration can be configured globally using the `MAT_TABS_CONFIG` injection token.
                "region": "slow-animation-duration"}) -->
 
 ### Accessibility
-`<mat-tab-group>` and `<mat-nav-tab-bar>` use different interaction patterns. The
-`<mat-tab-group>` component combines `tablist`, `tab`, and `tabpanel` into a single component with
-the appropriate keyboard shortcuts. The `<mat-nav-tab-bar>`, however, use a _navigation_ interaction
-pattern by using a `<nav>` element with anchor elements as the "tabs". The difference
-between these two patterns comes from the fact one updates the page URL while the other does not.
+`MatTabGroup` and `MatTabNavBar` imlement different interaction patterns for different use-cases.
```
Comment (crisbeto) [inline src/material/tabs/tabs.md:88]: imlement -> implement

File: src/material/tabs/tabs.md [hunk 4]
```diff
@@ -85,22 +85,36 @@ duration can be configured globally using the `MAT_TABS_CONFIG` injection token.
...
+You should choose the component that works best for your application.
+
+`MatTabGroup` combines `tablist`, `tab`, and `tabpanel` into a single component with
+handling forkeyboard inputs and focus management. You should use this component for switching
```
Comment (crisbeto) [inline src/material/tabs/tabs.md:92]: forkeyboard -> for keyboard

**Review by crisbeto — APPROVED** [review]
LGTM with some nits.

**Comment by aikidave** [comment]
I've added @TMDavisGoogle to provide an edit pass on this content. Please give her a day or two to review.

File: src/material/autocomplete/autocomplete.md (hunk 1 above)
Comment (TMDavisGoogle) [reply src/material/autocomplete/autocomplete.md]: id should be ID

File: src/material/tabs/tabs.md (hunk 3 above)
Comment (TMDavisGoogle) [reply src/material/tabs/tabs.md]: The `<mat-nav-tab-bar>`, however, use a _navigation_ interaction

Should be "uses a_navigation_interaction"

File: src/material/autocomplete/autocomplete.md [hunk 5]
```diff
@@ -112,8 +114,16 @@ autocomplete is attached to using the `matAutocompleteOrigin` directive together
               "region":"mat-autocomplete"}) -->
 
 ### Accessibility
-The input for an autocomplete without text or labels should be given a meaningful label via
-`aria-label` or `aria-labelledby`.
 
-The autocomplete trigger is given `role="combobox"`. The trigger sets `aria-owns` to the
-autocomplete's id, and sets `aria-activedescendant` to the active option's id.
```
Comment (TMDavisGoogle) [inline src/material/autocomplete/autocomplete.md:119]: id should be ID.

File: src/material/autocomplete/autocomplete.md [hunk 6]
```diff
@@ -112,8 +114,16 @@ autocomplete is attached to using the `matAutocompleteOrigin` directive together
...
+`MatAutocomplete` implements the ARIA combobox interaction pattern. The text input trigger specifies
+`role="combobox"` while the content of the pop-up applies `role="listbox"`. Because of this listbox
+pattern, you should _not_ put other interactive controls, such as buttons or checkboxes, inside of
```
Comment (TMDavisGoogle) [inline src/material/autocomplete/autocomplete.md:120]:
```suggestion
pattern, you should _not_ put other interactive controls, such as buttons or checkboxes, inside of
```
Should be "inside... ' (remove the "of")

File: src/material/bottom-sheet/bottom-sheet.md [hunk 7]
```diff
@@ -96,20 +96,49 @@ for `MAT_BOTTOM_SHEET_DEFAULT_OPTIONS` in your application's root module.
...
-In order to prevent users from tabbing into elements in the background, the Material bottom sheet
-uses a [focus trap](https://material.angular.io/cdk/a11y/overview#focustrap) to contain focus
-within itself. Once a bottom sheet is closed, it will return focus to the element that was focused
-before it was opened.
+`MatBottomSheet` creates modal dialogs that implement the ARIA `role="dialog"` pattern. This root
+dialog element should be given an accessible label via the `ariaLabel` property of
+`MatBottomSheetConfig`.
+
+#### Keyboard interaction
+By default, the escape key closes `MatBottomSheet`. While you can disable this behavior via
```
Comment (TMDavisGoogle) [inline src/material/bottom-sheet/bottom-sheet.md:105]:
```suggestion
By default, the escape key closes `MatBottomSheet`. While you can disable this behavior via
```
Remove "via". Sentence should read, "While you can disable this behavior by using..."

File: src/material/tabs/tabs.md [hunk 8]
```diff
@@ -85,22 +85,36 @@ duration can be configured globally using the `MAT_TABS_CONFIG` injection token.
                "region": "slow-animation-duration"}) -->
 
 ### Accessibility
```
Comment (twerske) [inline src/material/tabs/tabs.md:87]: nit: space needed?

**Comment by twerske** [comment]
out of scope, but have we thought about moving these to outside the API tab or making them more searchable for SEO and generally for developers looking at components?

File: src/material/tabs/tabs.md (hunk 8 above)
Comment (jelbourn) [reply src/material/tabs/tabs.md:87]: Do you mean the blank line?

**Comment by jelbourn** [comment]
@twerske these sections all live in the Overview tab for each component:
https://material.angular.io/components/icon/overview#accessibility

Did you have something in mind for SEO? AFAIK it should be indexed the same as the rest of the content.

**Pull request merged**